### PR TITLE
allow dynamically create PLANS_VALIDATORS

### PR DESCRIPTION
**Pull request by PetrDlouhy**
This enables PLANS_VALIDATORS be dynamically generated through lazy imported dict defined throught string.

### CODE REVIEW

File: plans/admin.py [hunk 1]
```diff
@@ -151,8 +157,8 @@ class UserPlanAdmin(UserLinkMixin, admin.ModelAdmin):
     list_display_links = list_display
     list_select_related = True
     readonly_fields = ['user_link', ]
-    fields = ('user_link', 'plan', 'expire', 'active' )
```
Comment (Alir3z4) [inline plans/admin.py]: Any reason why they are being converted to lists ?

File: plans/importer.py [hunk 2]
```diff
@@ -1,4 +1,8 @@
 def import_name(name):
-    components = name.split('.')
-    mod = __import__('.'.join(components[0:-1]), globals(), locals(), [components[-1]] )
-    return getattr(mod, components[-1])
+    """ import module given by str or pass the module if it is not str """
+    if type(name) == str:
```
Comment (Alir3z4) [inline plans/importer.py]: Couldn't this be written as `isinstance(name, str)` ?

File: plans/models.py [hunk 3]
```diff
@@ -429,6 +444,16 @@ class Order(models.Model):
     created = models.DateTimeField(_('created'), db_index=True)
     completed = models.DateTimeField(
         _('completed'), null=True, blank=True, db_index=True)
+    plan_extended_from = models.DateField(
+        _('The plan was extended from this date'),
+        null=True,
+        blank=True,
+    )
+    plan_extended_until = models.DateField(
+        _('The plan was extended until this date'),
```
Comment (Alir3z4) [inline plans/models.py]: This is not the right way of using `verbose_name`, this is more suited for `help_text`.

File: plans/models.py [hunk 4]
```diff
@@ -429,6 +444,16 @@ class Order(models.Model):
     created = models.DateTimeField(_('created'), db_index=True)
     completed = models.DateTimeField(
         _('completed'), null=True, blank=True, db_index=True)
+    plan_extended_from = models.DateField(
+        _('The plan was extended from this date'),
```
Comment (Alir3z4) [inline plans/models.py]: This is not the right way of using `verbose_name`, this is more suited for `help_text`.

**Comment by Alir3z4** [comment]
@PetrDlouhy just left some comments.
Also couldn't the migrations be done in one migration file instead of multiple of them ?

File: plans/admin.py (hunk 1 above)
Comment (PetrDlouhy) [reply plans/admin.py]: Changed back in #80

File: plans/models.py (hunk 3 above)
Comment (PetrDlouhy) [reply plans/models.py]: Fixed in #77

File: plans/models.py (hunk 4 above)
Comment (PetrDlouhy) [reply plans/models.py]: Fixed in #77

File: plans/importer.py (hunk 2 above)
Comment (PetrDlouhy) [reply plans/importer.py]: Fixed

**Comment by Alir3z4** [comment]
I see you have updated the code in other pull requests.
Is this pull request still valid to be merged ?

**Comment by Alir3z4** [comment]
Let me know when to merge this.

**Comment by PetrDlouhy** [comment]
I hope, this should be ready to merge.

**Comment by Alir3z4** [comment]
Awesome!
Merging now.

**Pull request merged**